### PR TITLE
Added Universal Sentence Encoder

### PR DESCRIPTION
**Pull request by MrPranav101**

### CODE REVIEW

File: models/use/__init__.py [hunk 1]
```diff
@@ -1,35 +1,56 @@
...
-
-EMBEDDING_MODELS: Dict[str, Embedding] = {embedding.name: embedding for embedding in EMBEDDING_MODELS}
\ No newline at end of file
+
+sess = tf.Session()
+sess.run([tf.global_variables_initializer(), tf.tables_initializer()])
+
+class Embeddings(object):
+    EMBEDDING_MODELS: List[Embedding] = [
+                        Embedding(name=u'use',
```
Comment (amansrivastava17) [inline models/use/__init__.py]: rename this `use_dan`

File: models/use/__init__.py [hunk 2]
```diff
@@ -1,35 +1,56 @@
...
+                                  dimensions=512,
+                                  corpus_size='na',
+                                  vocabulary_size='230k',
+                                  download_url='https://tfhub.dev/google/universal-sentence-encoder/2',
+                                  format='.tar.gz',
+                                  architecture='DAN',
+                                  trained_data='wikipedia and other sources',
+                                  language='en'),
+                        Embedding(name=u'use_large',
```
Comment (amansrivastava17) [inline models/use/__init__.py]: rename this to `use_transformer_large`

File: models/encode.py [hunk 3]
```diff
@@ -89,5 +89,16 @@ def _load_model(self):
...
-    def encode(self, text: str, pooling: str, **kwargs):
-        return self.embedding_cls.encode(text, pooling, **kwargs)
+    def encode(self, texts: Union[list, str], batch_size: int, pooling: str, **kwargs):
+        if type(texts) == str:
+            return self.embedding_cls.encode([texts], pooling, **kwargs)
+        else:
+            batch_opt = []
+            batch_tem = []
+            for i, text in enumerate(texts):
+                batch_tem.append(self.embedding_cls.encode([text], pooling, **kwargs))
```
Comment (chiragjn) [inline models/encode.py]: Don't use for loop, vectorize this operation

File: models/use/__init__.py [hunk 4]
```diff
@@ -1,35 +1,59 @@
...
+    model: str
+
+    @classmethod
+    def load_model(cls, model: str, model_path: str):
+        cls.use_module = hub.Module(model_path)
+        cls.model = model
+
+    @classmethod
+    def encode(cls, texts: list, pooling: str = None) -> Optional[np.array]:
+        sess.run([tf.global_variables_initializer(), tf.tables_initializer()])
```
Comment (chiragjn) [inline models/use/__init__.py]: This is a super heavy step. This should be run only once in a session that is an attribute of the instance of that `Embedding`

**Pull request merged**